pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -88,11 +88,16 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 		return framework.NewStatus(framework.Error, "node not found")
 	}
 	nodeConstraints := make(map[string]string)
+	nodeZoned := false
+	pvZoned := false
 	for k, v := range node.ObjectMeta.Labels {
 		if !volumeZoneLabels.Has(k) {
 			continue
 		}
 		nodeConstraints[k] = v
+		if k == v1.LabelZoneFailureDomain {
+			nodeZoned = true
```

Comment:
actually this does mean current node is zoned, for a non-zone node, it would be `failure-domain.beta.kubernetes.io/zone: "0"` and for a zoned node, it would be `failure-domain.beta.kubernetes.io/zone: eastus2-3`. 
So in mixed node pools scenario(user has both zoned and non-zoned node pools), let's ask user to add affinity on the pod/statefulset/deployoment.

Reply:
thx Andy! will it be too tricky if adding specific filter to filter the nodes with 'fault domain'/non-zoned label?

Reply:
I would suggest this: https://github.com/kubernetes/kubernetes/issues/97080#issuecomment-751203190

Reply:
got it. thx!